x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,218 @@
+	Name     string         `json:"name,omitempty"`
+	Surname  string         `json:"surname,omitempty"`
+	Bio      string         `json:"bio,omitempty"`
+	Pictures *Pictures      `json:"pictures,omitempty"`
+	Creator  sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgCreateProfile is a constructor function for MsgCreateProfile
+func NewMsgCreateProfile(name string, surname string, moniker string, bio string, pictures *Pictures,
+	creator sdk.AccAddress) MsgCreateProfile {
+	return MsgCreateProfile{
+		Moniker:  moniker,
+		Name:     name,
+		Surname:  surname,
+		Bio:      bio,
+		Pictures: pictures,
+		Creator:  creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCreateProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgCreateProfile) Type() string { return ActionCreateProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgCreateProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(msg.Name) != 0 && len(msg.Name) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) != 0 && len(msg.Surname) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(strings.TrimSpace(msg.Moniker)) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Profile moniker cannot be blank or empty")
+	}
+
+	if len(msg.Moniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.Bio) > MaxBioLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile biography cannot exceed %d characters", MaxBioLength))
+	}
+
+	return nil
+}
+
+// GetSignBytes encodes the message for signing
+func (msg MsgCreateProfile) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgCreateProfile) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Creator}
+}
+
+// ----------------------
+// --- MsgEditProfile
+// ----------------------
+
+// MsgEditProfile defines a EditProfile message
+type MsgEditProfile struct {
+	PreviousMoniker string         `json:"previous_moniker"`
```

Comment:
Probably we can remove this as we should be able to fetch the profile from the address only